### ISSUE TITLE
False positive on ImplementEqualsHashCodeOnValueObjects

### ISSUE THREAD

**Issue by rodrigolopes** [jborgers/PMD-jPinpoint-rules#175]
The following class fails validation with ImplementEqualsHashCodeOnValueObjects rule:
(A java VO annotated with lombok @Getter and @Setter and with hashCode() and equals() implementations)

```
package vo;

import java.util.Objects;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class FailingPMDRule {

    private String prop1;
    private String prop2;
    private String prop3;
    private String prop4;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        FailingPMDRule that = (FailingPMDRule) o;
        return Objects.equals(prop1, that.prop1) && Objects.equals(prop2, that.prop2) && Objects.equals(prop3, that.prop3) && Objects.equals(prop4, that.prop4);
    }

    @Override
    public int hashCode() {
        return Objects.hash(prop1, prop2, prop3, prop4);
    }
}

```
